contracts/modules/uniswap/v2/UniswapV2Library.sol

```diff
@@ -52,15 +52,22 @@ library UniswapV2Library {
         pure
         returns (address pair)
     {
-        pair = address(
-            uint160(
-                uint256(
-                    keccak256(
-                        abi.encodePacked(hex'ff', factory, keccak256(abi.encodePacked(token0, token1)), initCodeHash)
-                    )
-                )
-            )
-        );
+        assembly ("memory-safe") {
```

Comment:
I think I like the gas savings here. Wonder if it's worth a comment:

```
/// accomplishes the following with with clean upper bits
///uint160(keccak256(abi.encodePacked(hex'ff', factory, keccak256(abi.encodePacked(token0, token1)), initCodeHash)))
```
to get a feel for what this is doing at a glance

Reply:
Added.

Reply:
thanks,  I think that's pretty helpful for readability, one for the V3Lib would probably make sense too. (sorry, missed that)

Reply:
Added.